### PR TITLE
docs: language select and outdated docs banner

### LINKED ISSUE #770: feat: in translated docs, show if the English version is newer
### Is your feature request related to a problem? Please describe.

With Arabic and Russian docs translations underway, we should start thinking about how to integrate the docs well. 

One common issue with docs translations is that they become out of date as the English version gets more updates. Explicitly showing that could give people a warning that there might be newer information, and maybe also motivate more people to contribute to translations. 

### Describe the solution you'd like to see

A component that shows up on translated docs if the English version of this page is newer than the translated version. 

Something like “This translation was last updated on (date). Since then there have been (number) revisions to the English version, with the newest one being on (date). Click here to see the English version of this page. Or click here to help us update the translated version” (I’m sure there is a better way to write this)

Of course this component would ideally be translated into each of the languages we support in the docs. 

### Describe alternate solutions

Not doing this

### Additional information

I don’t think it’s super urgent as all the incoming translations will be new for the time being, but it would be an awesome thing to have.

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes #770

## ✅ Checklist

- [ ] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [ ] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] I performed a functional test on my final commit

---

## Changelog

_[Short description of what has changed]_

---

## Screenshots

_[Screenshots]_

💯

### CODE REVIEW

**Comment by juliusmarminge** [comment]
CleanShot 2022-11-17 at 21 11 15@2x

needs some more styling and transitions

**Comment by juliusmarminge** [comment]
@c-ehrlich I'm not sure how we're supposed to show how many revisions behind the English version the page is? Right now just fetching a single commit and comparing their dates.

What would be considered a revision?

**Comment by juliusmarminge** [comment]
This is how it looks:

CleanShot 2022-11-17 at 22 28 53@2x

**Comment by juliusmarminge** [comment]
Left todo:

- Animate opening?
- Close when clicking outside

**Comment by c-ehrlich** [comment]
The header is currently overflowing on mobile

image

The Vue docs (imo one of the best internationalizations) just uses the language icon without showing which language you're currently on. I think that's ok, the active language is somewhat self evident information (it's either the one you want to read the docs in, or it isn't)
image

Banner looks good from screenshots. Will take a look at the code tonight.

**Comment by nexxeln** [comment]
> The Vue docs (imo one of the best internationalizations) just uses the language icon without showing which language you're currently on. I think that's ok, the active language is somewhat self evident information (it's either the one you want to read the docs in, or it isn't)
 Yep I agree. It is taking too much space too. Also I don't think we need to generalize the languages with country flags. Just the language name is fine

**Comment by hodeinavarro** [comment]
Astro also includes a note block when a page is not translated at all, could this be added too?
image

> Note
This page is not available on your language yet, meanwhile you will see the english version. Help us translate! More information about how to contribute.

**Comment by juliusmarminge** [comment]
> Yep I agree. It is taking too much space too. Also I don't think we need to generalize the languages with country flags. Just the language name is fine

I like flags :( Fair point about the generalization but I still think it adds something

**Comment by juliusmarminge** [comment]
We should probably have some fallback behavior if the files doesn't exist yet for a specific langauge right?

https://user-images.githubusercontent.com/51714798/202825138-10f76736-6c8f-4a08-a0e3-18bcf45fe1fe.mp4

Example: Astro renders the english version in the sidebar if it doesn't exist for the native language:

CleanShot 2022-11-19 at 01 44 56@2x

File: www/src/pages/sv/installation.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
```
Comment (juliusmarminge) [inline www/src/pages/sv/installation.md:1]: NOTE: These files should be removed before merging

File: www/src/components/docs/outdatedDocsBanner.astro [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+{
+  diffInDays > 0 && (
+    <div class="mb-8 w-full rounded-lg bg-t3-purple-200/50 p-4 font-medium dark:bg-t3-purple-200/10">
+      <p>
+        Attention: This page is{" "}
+        <span class="font-extrabold">
+          {diffInDays} {diffInDays === 1 ? "day" : "days"}
+        </span>
+        older than the English version, and might be outdated. If you're a
```
Comment (c-ehrlich) [inline www/src/components/docs/outdatedDocsBanner.astro:51]: could this take a `language: KnownLanguage` prop (or some other way to know what the active language is) and then render the message in that language, with a fallback to English?

Comment (c-ehrlich) [reply www/src/components/docs/outdatedDocsBanner.astro:51]: oh probably just `const [_1, _2, lang] = path.split("/");`

File: www/src/components/docs/outdatedDocsBanner.astro [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        native speaker of this language, and would like to contribute to the
+        project, please consider updating this page to match the latest English
+        version.
+      </p>
+      <p class="mt-3">
+        You can also{" "}
+        <a class="text-t3-purple-400" href={engHref}>
+          to go to the English version
```
Comment (AyanavaKarmakar) [inline www/src/components/docs/outdatedDocsBanner.astro:59]:
```suggestion
          go to the English version
```

File: www/src/components/docs/outdatedDocsBanner.astro (hunk 2 above)
Comment (AyanavaKarmakar) [inline www/src/components/docs/outdatedDocsBanner.astro:51]:
```suggestion
        older than the English version and might be outdated. If you're a
```

**Review by AyanavaKarmakar — CHANGES_REQUESTED** [review]

File: www/src/components/docs/outdatedDocsBanner.astro [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+{
+  diffInDays > 0 && (
+    <div class="mb-8 w-full rounded-lg bg-t3-purple-200/50 p-4 font-medium dark:bg-t3-purple-200/10">
+      <p>
+        Attention: This page is{" "}
+        <span class="font-extrabold">
+          {diffInDays} {diffInDays === 1 ? "day" : "days"}
+        </span>
+        older than the English version, and might be outdated. If you're a
+        native speaker of this language, and would like to contribute to the
```
Comment (AyanavaKarmakar) [inline www/src/components/docs/outdatedDocsBanner.astro:52]:
```suggestion
        native speaker of this language and would like to contribute to the
```

File: www/src/components/docs/outdatedDocsBanner.astro (hunk 3 above)
Comment (juliusmarminge) [inline www/src/components/docs/outdatedDocsBanner.astro:59]:
```suggestion
        view the English version
```

**Review by c-ehrlich — CHANGES_REQUESTED** [review]
remove the nonexistent test languages and we're ready to merge :)

**Review by c-ehrlich — APPROVED** [review]
💯

**Pull request merged**